tests/compiler/test_typing.py

```diff
@@ -159,27 +159,85 @@ def test_cast_to_number(snippetcompiler):
     assert Number().validate(u)
 
 
-def test_compare_float_int_attribute(snippetcompiler):
+def test_same_value_float_int(snippetcompiler):
     snippetcompiler.setup_for_snippet(
         """
-    entity Test:
-        float i = 0
+    i = 42.0
+    j = 42
+    i = j
+    j = i
+    """,
+    )
+    (_, scopes) = compiler.do_compile()
+
+
+def test_different_value_float_int(snippetcompiler, capsys):
+    snippetcompiler.setup_for_error(
+        """
+    i = 42.1
+    j = 42
+    i = j
+    """,
+        "value set twice:\n"
+        "\told value: 42.1\n"
+        "\t\tset at {dir}/main.cf:2\n"
+        "\tnew value: 42\n"
+        "\t\tset at {dir}/main.cf:4:9\n"
+        " (reported in i = j ({dir}/main.cf:4))",
+    )
+
+
+@pytest.mark.parametrize("val", ["42.0", "42.1"])
+def test_float_int_attribute(snippetcompiler, val):
+    snippet = f"""
+    entity Int:
+        int i
     end
 
-    entity Test2:
-        int i = 0
+    entity Float:
+        float i
     end
 
-    implement Test using std::none
-    implement Test2 using std::none
+    implement Int using std::none
+    implement Float using std::none
 
-    val1 = Test(i = 42.0)
-    val2 = Test2(i = 42)
-    val2.i = val1.i
-    val1.i = val2.i
-    """,
+    f = Float(i={val})
```

Comment:
This part of the test is not really stable. If the `Int` constructor is executed first the compile fails before we ever execute this statement.